File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.rb
# @param {Integer[]} nums
# @return {Integer}
def maximum_difference(nums)
    max = -1
    nums.each_index do |i|
        (i+1...nums.size).each do |j|
            diff = nums[j] - nums[i]
            max = diff if diff > max  && diff > 0
        end
    end
    max
end

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int n=nums.size();
        int maxdiff = -1;

        for (int i=0;i<n;++i) {
            for (int j=i+1;j<n;++j) {
                int diff = nums[j] - nums[i];
                if (diff > 0) {
                    maxdiff=max(maxdiff,diff);
                }
            }
        }
        return maxdiff;   
    }
};